### ISSUE TITLE
Check processing of BBS data into communities

### ISSUE THREAD

**Issue by ha0ye** [weecology/MATSS#165]
Currently, the processing of BBS data splits it into separate communities, which should correspond to different routes. I think @diazrenata might have grabbed code from `weecology/bbs-forecasting` for this.

It looks like just `route` and `bcr` are insufficient to fully separate out routes. For example, I see:
```
  countrynum statenum route    routename active latitude  longitude stratum bcr routetypeid  routetypedetailid   regionname
1        124        4     1 SEVEN PERSON      0 49.93222 -110.84452      38  11           1                  1      Alberta
2        840       64     1       FORMAN      1 46.15318  -97.67468      37  11           1                  1 NORTH DAKOTA
3        124       79     1     CARNDUFF      1 49.17711 -101.84405      30  11           1                  1 Saskatchewan
4        840       81     1       LENNOX      1 43.33062  -96.88568      37  11           1                  1 SOUTH DAKOTA
```

So we may need to additional filter for `countrynum` and `statenum` as well?

**Comment by diazrenata** [comment]
Yep, the code for handling BBS is based on the code in bbsforecasting. Good catch!!

**Comment by ha0ye** [comment]
> Yep, the code for handling BBS is based on the code in bbsforecasting. Good catch!!

I took a quick look in that repo, but I didn't see anything on splitting of BBS into community-level units?

**Comment by diazrenata** [comment]
:thinking_face: Pretty sure this is a mistake on my end. 

I based the data handling on bbs-forecasting, mostly for dealing with subspecies and filtering. I introduced the subsetting structure because some commands are super slow on the whole data frame at once. Digging in, it looks like bbs-forecasting identifies routes using *state number* and route, not *region* and route. See https://github.com/weecology/bbs-forecasting/blob/master/R/forecast-bbs-core.R <https://github.com/weecology/bbs-forecasting/blob/master/R/forecast-bbs-core.R> @ line 156. Routes are definitely the “community” unit for that study (https://www.biorxiv.org/content/biorxiv/early/2017/12/11/191130.full.pdf <https://www.biorxiv.org/content/biorxiv/early/2017/12/11/191130.full.pdf>). 

I could sleuth more, but given what I know about my own knowledge of BBS, I’m confident I just thought (wrongly) that region + route was the appropriate combination. Thanks for noticing this!!

**Comment by ha0ye** [comment]
Ah ok, that makes sense. Thanks for digging out the answer!

I can merge in a fix in my next set of updates.

**Comment by ha0ye** [comment]
To be fair, `statenum` is the column name for the variable that contains "Region ID number", and this is different from `bcr`, which is "Bird Conservation Region". So it is still routes and region, but a different "region"...

¯\_(ツ)_/¯

**Comment by diazrenata** [comment]
🙃 this makes me feel a little better!

**Comment by ha0ye** [comment]
closed in 095c6f5
